### ISSUE TITLE
Overlay with merge_page doesn't work for a certain overlay PDF file

### ISSUE THREAD

**Issue by Andras1000** [py-pdf/pypdf#2301]
Following the documentation here: https://pypdf.readthedocs.io/en/latest/user/add-watermark.html#watermark-overlay

I want to overlay one PDF over another. The below code works for a certain more complex `overlay.pdf`, but doesn't work for the simple `overlay.pdf` linked below. I see the base PDF, but not the overlay.

## Environment

Tested both on Ventura 13.4 and Ubuntu 22.04
Python 3.10
pypdf 3.16.4 and 3.17.1

```bash
$ python -m platform
macOS-10.16-x86_64-i386-64bit
Linux-6.2.0-1015-aws-x86_64-with-glibc2.35


$ python -c "import pypdf;print(pypdf._debug_versions)"
pypdf==3.17.1, crypt_provider=('cryptography', '41.0.1'), PIL=10.0.0 (Mac)
pypdf==3.17.1, crypt_provider=('cryptography', '3.4.8'), PIL=none (Ubuntu)
```

## Code + PDF

This is a minimal, complete example that shows the issue:

```
from pypdf import PdfWriter, PdfReader

stamp = PdfReader("overlay.pdf").pages[0]
writer = PdfWriter(clone_from="form_1099_nec_page_4.pdf")
for page in writer.pages:
    page.merge_page(stamp)

writer.write("merged.pdf")
```

https://www.dropbox.com/scl/fo/ygqttuti4ed7cc2dz1c4u/h?rlkey=khdu0k5stkt205p4mcgca5n0i&dl=0

This `overlay.pdf` was generated with the following code, in case that's relevant.

```
wkhtmltopdf_path = getattr(settings, 'WKHTMLTOPDF_PATH', None)
config = pdfkit.configuration(wkhtmltopdf=wkhtmltopdf_path)
with open('overlay.html', 'r') as html_file:
    html_content = html_file.read()

pdf_data = pdfkit.from_string(html_content, False, configuration=config)

with open('overlay.pdf', 'wb') as pdf_file:
    pdf_file.write(pdf_data)
```

## Traceback

There is no error, but the I only see the base PDF without the overlay when opening `merged.pdf`. It does work with another overlay PDF file though.

**Comment by Andras1000** [comment]
I managed to solve the issue by first converting the base PDF (`form_1099_nec_page_4.pdf`) to JPG, then converting it back to PDF.

The issue is potentially with XFA and the above process could be removing the XFA layer.

**Comment by stefan6419846** [comment]
Converting the PDF to JPG and converting it back to PDF will usually leave you with a PDF which has a full-page image file only, thus mostly removing all "interesting" stuff and dropping the vector features. While it might work in your case, I still consider this a possible bug on the *pypdf* side.

**Comment by Andras1000** [comment]
> Converting the PDF to JPG and converting it back to PDF will usually leave you with a PDF which has a full-page image file only, thus mostly removing all "interesting" stuff and dropping the vector features. While it might work in your case, I still consider this a possible bug on the _pypdf_ side.

Good point! Somehow it started to work. I don't remember what I did. At one point I tried to swap which file is the overlay and which file is the stamp.

**Comment by pubpub-zz** [comment]
@Andras1000 
can you try to add `expand=True` extra parameter to `merge_page` ?

**Comment by Andras1000** [comment]
In my case, the PDF I was overlaying on top of another PDF was generated from HTML using `wkhtmltopdf`. Upgrading from "wkhtmltopdf 0.12.6" to "wkhtmltopdf 0.12.6 (with patched qt)" seems to have solved the issue (Ubuntu 22.04).
